### PR TITLE
Fix and unit test for issue #1275

### PR DESCRIPTION
**Pull request by SamCooper**
Hello, after hitting this issue in Property Sheets I thought I could take the time to create the fix based on the suggestion in issue #1275 and create the unit test for it too.

### CODE REVIEW

File: controlsfx/src/main/java/org/controlsfx/property/editor/Editors.java [hunk 1]
```diff
@@ -98,6 +98,7 @@ private Editors() {
 
             @Override public void setValue(Number value) {
                 sourceClass = (Class<? extends Number>) value.getClass();
+                ((NumericField)getEditor()).setValue(value);
```
[the remark this answers is not in the document]
Comment (eugener) [reply controlsfx/src/main/java/org/controlsfx/property/editor/Editors.java:101]: Would it be better to listen for text changes within the `NumericField` and set the value internally on text change without exposing `value`.

File: controlsfx/src/test/java/org/controlsfx/property/editor/NumericPropertyZeroTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 20121, ControlsFX
```
Comment (Maxoudela) [inline controlsfx/src/test/java/org/controlsfx/property/editor/NumericPropertyZeroTest.java:2]: Typo

File: controlsfx/src/main/java/org/controlsfx/property/editor/NumericField.java [hunk 3]
```diff
@@ -77,6 +77,10 @@ public NumericField( Class<? extends Number> cls ) {
         }
     }
 
+    protected void setValue(Number v) {
```
Comment (Maxoudela) [inline controlsfx/src/main/java/org/controlsfx/property/editor/NumericField.java:80]: If we do keep this method (see @eugener comment), it will need a javadoc.

**Review by Maxoudela — CHANGES_REQUESTED** [review]

File: controlsfx/src/main/java/org/controlsfx/property/editor/Editors.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (SamCooper) [reply controlsfx/src/main/java/org/controlsfx/property/editor/Editors.java:101]: I don't think you do need to check the cast as you've passed it into AbstractPropertyEditor, and set the generic type of AbstractPropertyEditor to NumericField.

If fact the code at that line should just be:

`getEditor().setValue(value);`

without the cast (compiles and works fine actually).

I tried the listener approach and it occasionally resulted in `java.lang.IllegalArgumentException: The start must be <= the end` I think caused by an empty string. It was difficult to replicate as some sort of race condition was happening.




Anyway, TLDR, I think the cast can be removed and all is ok. I'll push an update to reflect this.

File: controlsfx/src/test/java/org/controlsfx/property/editor/NumericPropertyZeroTest.java (hunk 2 above)
Comment (SamCooper) [reply controlsfx/src/test/java/org/controlsfx/property/editor/NumericPropertyZeroTest.java]: Fixed

File: controlsfx/src/main/java/org/controlsfx/property/editor/NumericField.java (hunk 3 above)
Comment (SamCooper) [reply controlsfx/src/main/java/org/controlsfx/property/editor/NumericField.java]: Fixed

**Review by Maxoudela — APPROVED** [review]

**Comment by abhinayagarwal** [comment]
Hi @SamCooper,

Can you please update the license header in the 2 files changed to "Copyright (c) 2015, 2021, ControlsFX" ?

**Review by eugener — APPROVED** [review]

**Review by abhinayagarwal — APPROVED** [review]
